Alicia Sigmon

als333@pitt.edu

10/17/2017

# Discourse Analysis of the Australian Radio Talkback Courpus

#### About the Corpus
- The Australian Radio Talkback corpus contains raw text files of telephone conversations.
- These conversations include the speaker's role (presenter, expert, or caller), name, and gender.
- The conversations include other verbal cues such as laughter <laugh> and where a speaker says something during another speaker's turn <E1 yeah>. It also notes corrections to the transcription in squirrley brackets {}.

#### Discourse Analysis Goals
- commparing speakers by role and gender
- aspects to consider:
    - back channels (yeah, uhhu, that's great!)
    - vocabulary size (avg word length)
    - number of turns, sentences, and words

In [1]:
%pprint

Pretty printing has been turned OFF


In [2]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [3]:
from nltk.corpus import PlaintextCorpusReader
import pandas as pd
import nltk

In [4]:
# Getting the fileids for ART
# Using PlaintextCorpusReader
ART_corpus_root = r'C:\Users\sigmo\Documents\Data_Science\Project-Alicia\data_files\AustralianRadioTalkback\files\Raw'
ART_corpus = PlaintextCorpusReader(ART_corpus_root, '.*')

ART_fids=[x for x in ART_corpus.fileids()]

In [5]:
len(ART_fids)
ART_fids[:5]

26

['ABCE1-raw.txt', 'ABCE2-raw.txt', 'ABCE3-raw.txt', 'ABCE4-raw.txt', 'ABCNE1-raw.txt']

## Mini Test

In [6]:
ART_corpus.raw()[:1000]==ART_corpus.raw('ABCE1-raw.txt')[:1000]
ART_corpus.raw()[:1000]
# Each new line begins with the speaker in [] and ends with \n\n

True

"[Presenter 1: Simon Marnie, M] Thanks for that John Hall now John Hall will be listening for the next hour 'cos Angus Stewart is here to take your calls eight-triple-three-one-thousand one-eight-hundred-eight-hundred-seven-oh-two something in the garden that's causing you problems give us a call right now and Angus can I mean y'know he is known in the trade as Mr popergation {propagation} Mr propagation. He's also known for his passion for natives and his love of o orchids am I right so far.\n\n[Expert 1: Angus Stewart, M] I guess yeah yeah <laughs>.\n\n[P1] He's also known <E1 sounds reasonable> for his ability to open cosposting {composting} toilets so he can tell you anything worm farm problems certainly helped us and although I'm still confused about dry ingredients we might talk about that as well but eight-triple-three-one-thousand one-eight-hundred-eight-hundred-seven-oh-two fine sunny day today top temperatures on the coast of twenty-seven inland thirty degrees Bowral enjoying

In [7]:
dir(ART_corpus.raw()[:1000])

mini=ART_corpus.raw()[:1000]
# list split by speakers' lines
mini_art=mini.split('\n\n')
mini_art

# 1st attempt at word tokenizing
mini_words=nltk.word_tokenize(mini)
mini_words
    # this does not keep the information of the speaker within brackets as one token
    # how can I separate this information?

['__add__', '__class__', '__contains__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__getnewargs__', '__gt__', '__hash__', '__init__', '__iter__', '__le__', '__len__', '__lt__', '__mod__', '__mul__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__rmod__', '__rmul__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'capitalize', 'casefold', 'center', 'count', 'encode', 'endswith', 'expandtabs', 'find', 'format', 'format_map', 'index', 'isalnum', 'isalpha', 'isdecimal', 'isdigit', 'isidentifier', 'islower', 'isnumeric', 'isprintable', 'isspace', 'istitle', 'isupper', 'join', 'ljust', 'lower', 'lstrip', 'maketrans', 'partition', 'replace', 'rfind', 'rindex', 'rjust', 'rpartition', 'rsplit', 'rstrip', 'split', 'splitlines', 'startswith', 'strip', 'swapcase', 'title', 'translate', 'upper', 'zfill']

["[Presenter 1: Simon Marnie, M] Thanks for that John Hall now John Hall will be listening for the next hour 'cos Angus Stewart is here to take your calls eight-triple-three-one-thousand one-eight-hundred-eight-hundred-seven-oh-two something in the garden that's causing you problems give us a call right now and Angus can I mean y'know he is known in the trade as Mr popergation {propagation} Mr propagation. He's also known for his passion for natives and his love of o orchids am I right so far.", '[Expert 1: Angus Stewart, M] I guess yeah yeah <laughs>.', "[P1] He's also known <E1 sounds reasonable> for his ability to open cosposting {composting} toilets so he can tell you anything worm farm problems certainly helped us and although I'm still confused about dry ingredients we might talk about that as well but eight-triple-three-one-thousand one-eight-hundred-eight-hundred-seven-oh-two fine sunny day today top temperatures on the coast of twenty-seven inland thirty degrees Bowral enjoyin

['[', 'Presenter', '1', ':', 'Simon', 'Marnie', ',', 'M', ']', 'Thanks', 'for', 'that', 'John', 'Hall', 'now', 'John', 'Hall', 'will', 'be', 'listening', 'for', 'the', 'next', 'hour', "'cos", 'Angus', 'Stewart', 'is', 'here', 'to', 'take', 'your', 'calls', 'eight-triple-three-one-thousand', 'one-eight-hundred-eight-hundred-seven-oh-two', 'something', 'in', 'the', 'garden', 'that', "'s", 'causing', 'you', 'problems', 'give', 'us', 'a', 'call', 'right', 'now', 'and', 'Angus', 'can', 'I', 'mean', "y'know", 'he', 'is', 'known', 'in', 'the', 'trade', 'as', 'Mr', 'popergation', '{', 'propagation', '}', 'Mr', 'propagation', '.', 'He', "'s", 'also', 'known', 'for', 'his', 'passion', 'for', 'natives', 'and', 'his', 'love', 'of', 'o', 'orchids', 'am', 'I', 'right', 'so', 'far', '.', '[', 'Expert', '1', ':', 'Angus', 'Stewart', ',', 'M', ']', 'I', 'guess', 'yeah', 'yeah', '<', 'laughs', '>', '.', '[', 'P1', ']', 'He', "'s", 'also', 'known', '<', 'E1', 'sounds', 'reasonable', '>', 'for', 'his', 'a

In [8]:
mini_art[0]
endbrack = mini_art[0].index(']')
print(mini_art[0][1:endbrack])
print(mini_art[0][endbrack+2:])

"[Presenter 1: Simon Marnie, M] Thanks for that John Hall now John Hall will be listening for the next hour 'cos Angus Stewart is here to take your calls eight-triple-three-one-thousand one-eight-hundred-eight-hundred-seven-oh-two something in the garden that's causing you problems give us a call right now and Angus can I mean y'know he is known in the trade as Mr popergation {propagation} Mr propagation. He's also known for his passion for natives and his love of o orchids am I right so far."

Presenter 1: Simon Marnie, M
Thanks for that John Hall now John Hall will be listening for the next hour 'cos Angus Stewart is here to take your calls eight-triple-three-one-thousand one-eight-hundred-eight-hundred-seven-oh-two something in the garden that's causing you problems give us a call right now and Angus can I mean y'know he is known in the trade as Mr popergation {propagation} Mr propagation. He's also known for his passion for natives and his love of o orchids am I right so far.


## Creating lists for data

In [9]:
# splitting all lines
ART_lines=ART_corpus.raw().split("\n\n")
ART_lines[:6]

["[Presenter 1: Simon Marnie, M] Thanks for that John Hall now John Hall will be listening for the next hour 'cos Angus Stewart is here to take your calls eight-triple-three-one-thousand one-eight-hundred-eight-hundred-seven-oh-two something in the garden that's causing you problems give us a call right now and Angus can I mean y'know he is known in the trade as Mr popergation {propagation} Mr propagation. He's also known for his passion for natives and his love of o orchids am I right so far.", '[Expert 1: Angus Stewart, M] I guess yeah yeah <laughs>.', "[P1] He's also known <E1 sounds reasonable> for his ability to open cosposting {composting} toilets so he can tell you anything worm farm problems certainly helped us and although I'm still confused about dry ingredients we might talk about that as well but eight-triple-three-one-thousand one-eight-hundred-eight-hundred-seven-oh-two fine sunny day today top temperatures on the coast of twenty-seven inland thirty degrees Bowral enjoyin

In [10]:
for y in ART_fids[:2]:
    print(y)   

ABCE1-raw.txt
ABCE2-raw.txt


In [11]:
# Taking the speaker information, text, and file name for each line
# and storing them in individual lists

# also counting the number of utterances per file to use as an index in art_df

speaker=[]
text=[]
line_fid=[]
utterance_num=[]

z=""

for y in ART_fids:
    count=0
    
    for x in ART_corpus.raw(y).split("\n\n"):
        count+=1
        
#         # Adjusting for NAT4.5-raw.txt:
#         if y=="NAT4-raw.txt":
#             if x.startswith("[Presenter 1: John Cleary, M]"):
#                 z="NAT4.5-raw.txt"
#             if z=="NAT4.5-raw.txt":
#                 if x.startswith("["):
#                     endbrack = x.index(']')
#                     line_fid.append(z)
#                     speaker.append(x[1:endbrack])
#                     text.append(x[endbrack+2:])
#                     utterance_num.append(count)
        
        if x.startswith("["):
            endbrack = x.index(']')
            line_fid.append(y)
            speaker.append(x[1:endbrack])
            text.append(x[endbrack+2:])
            utterance_num.append(count)

# text[3093]   # this line is actually 2 lines!! E1 and C4!
# line_fid[3093] # COME3



# This was attempted inside the 2nd for loop above to fix the line with the index 3093
# in order to fix the line shown above because [C4]'s line is contianed within [E1]'s line
    # there will be a better way to do this

# if count == 3093:
#     sent = "[E1] That sounds exactly <C4 inaudible> what it is just because you've had a hernia operation uh any sort of operation gall bladder operation hernia operation <,> doesn't mean that you can't get other things wrong with you and you still do you still get tonsillitis you still <C4 yeah> get appendicitis and you can still get gastroenteritis which sounds like what you've got.\r\n\r\n[C4] Nah <E1 it ih thi> and because of the operation that's stirring the my tummy up."
#     sent.split("\r\n\r\n")
#     for item in new_split:
#         endbrack = x.index(']')
#         line_fid.append(y)
#         speaker.append(x[1:endbrack])
#         text.append(x[endbrack+2:])
#         utterance_num.append(count)

In [12]:
# checking the lists
len(line_fid)
line_fid[:10]
line_fid[-10:]

len(speaker)
speaker[:10]

len(text)
text[:10]

len(utterance_num)
utterance_num[:1500]

9026

['ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt']

['NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt']

9026

['Presenter 1: Simon Marnie, M', 'Expert 1: Angus Stewart, M', 'P1', 'E1', 'P1', 'P1', 'Caller 1: Suzanne, F', 'P1', 'C1', 'P1']

9026

["Thanks for that John Hall now John Hall will be listening for the next hour 'cos Angus Stewart is here to take your calls eight-triple-three-one-thousand one-eight-hundred-eight-hundred-seven-oh-two something in the garden that's causing you problems give us a call right now and Angus can I mean y'know he is known in the trade as Mr popergation {propagation} Mr propagation. He's also known for his passion for natives and his love of o orchids am I right so far.", 'I guess yeah yeah <laughs>.', "He's also known <E1 sounds reasonable> for his ability to open cosposting {composting} toilets so he can tell you anything worm farm problems certainly helped us and although I'm still confused about dry ingredients we might talk about that as well but eight-triple-three-one-thousand one-eight-hundred-eight-hundred-seven-oh-two fine sunny day today top temperatures on the coast of twenty-seven inland thirty degrees Bowral enjoying twenty-seven and Katoomba twenty-five degrees currently around 

9026

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 22

In [13]:
count=0
for x in line_fid:
    if x.startswith("NAT4.5"):
        count+=1
count # 150 lines in the new file, NAT4.5

0

In [14]:
# speaker information compiled
speaker_infos=[]
speaker_infos_fid=[]
count=0
for x in speaker:
    y=line_fid[count] # trying to fix duplicates
    count+=1
    if len(x)>5:
        speaker_infos.append(x)
        speaker_infos_fid.append(y)

In [15]:
len(speaker_infos)
speaker_infos[:20]

len(speaker_infos_fid)
speaker_infos_fid[:20]

428

['Presenter 1: Simon Marnie, M', 'Expert 1: Angus Stewart, M', 'Caller 1: Suzanne, F', 'Caller 2: Lisa, F', 'Caller 3: Sally, F', 'Caller 4: Danny, M', 'Caller 5: Trevor, M', 'Caller 6: Gillian, F', 'Caller 7: Colleen, F', 'Caller 8: Bernie, M', 'Caller 9: William, M', 'Expert 2: Jeanne Villani, F', 'Caller 10: Beth, F', 'Caller 11: Lynne, F', 'Caller 12: Jack, M', 'Presenter 1: Simon Marnie, M', 'Expert 1: Les, M', 'Expert 2: Pete, M', 'Caller 1: Julie, F', 'Caller 2: Janelle, F']

428

['ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE2-raw.txt', 'ABCE2-raw.txt', 'ABCE2-raw.txt', 'ABCE2-raw.txt', 'ABCE2-raw.txt']

In [16]:
# simon marnie is presenter 1 for both files ABCE1 and ABCE2
count=0
for x in speaker_infos:
    if "Simon Marnie" in x:
        print(x) # he's only in those 2 files! 
        print(count)
    count+=1

speaker_infos_fid[0]
speaker_infos_fid[15] # shows he's in dif files <-- will work on fixing once all speaker info is in the same format!!

Presenter 1: Simon Marnie, M
0
Presenter 1: Simon Marnie, M
15


'ABCE1-raw.txt'

'ABCE2-raw.txt'

In [17]:
# finding formats missing the :
for x in speaker_infos:
    if ":" not in x:
        x
        speaker_infos.index(x)
        speaker.index(x)

# what's happening with Paul?

'Caller 11, Robyn, F'

40

1478

'Caller 23, Maureen, F'

98

2699

'Caller 9 Maureen, F'

121

3172

'P1b Paul Murray, M'

168

4483

'Caller 12, Brian, M'

408

8600

'Caller 10, Brett, M'

420

8858

In [18]:
# Examining file COME6-raw.txt
speaker.index('P1b Paul Murray, M') # 4483
speaker[4400:4500]

speaker.index('Presenter 1: Paul Murray, M') # 4436
line_fid[4436] #COME6-raw.txt

# after looking at this file, I think it needs to be removed!!
    # The document switches back and forth between P1a and P1b, and once P1a had an interruption in P1b

4483

['C24', 'P2', 'C24', 'P2', 'C24', 'P2', 'C24', 'P2', 'C24', 'P2', 'C24', 'P2', 'C24', 'P2', 'C24', 'P2', 'C24', 'P2', 'C24', 'P2', 'C24', 'P2', 'C24', 'P1', 'P2', 'C24', 'P1', 'C24', 'P2', 'C24', 'P2', 'P1', 'P2', 'P1', 'P2', 'P1', 'Presenter 1: Paul Murray, M', 'Caller 1: Ella, F', 'P1a', 'C1', 'P1a', 'C1', 'P1a', 'C1', 'P1a', 'Caller 2: TJ, M', 'P1a', 'C2', 'P1a', 'C2', 'P1a', 'Caller 3: Jo, F', 'P1a', 'C3', 'P1a', 'C3', 'P1a', 'C3', 'P1a', 'C3', 'P1a', 'Caller 4: Ruben, M', 'P1a', 'C4', 'P1a', 'C4', 'P1a', 'Caller 5: Johnny, M', 'P1a', 'C5', 'P1a', 'Caller 6: LJ, M', 'P1a', 'C6', 'P1a', 'C6', 'P1a', 'C6', 'P1a', 'C6', 'P1a', 'C6', 'P1a', 'P1b Paul Murray, M', 'Caller 7: Megan, F', 'P1b', 'C7', 'P1b', 'C7', 'P1b', 'C7', 'P1b', 'C7', 'P1b', 'C7', 'P1b', 'C7', 'P1b', 'C7', 'P1b']

4436

'COME6-raw.txt'

In [19]:
# manually adding the colon for first time info 
speaker_infos[40]='Caller 11: Robyn, F'
speaker_infos[98]='Caller 23: Maureen, F'
speaker_infos[121]='Caller 9: Maureen, F'
speaker_infos[168]='Presenter 1b: Paul Murray, M' # probably deleting file
speaker_infos[408]='Caller 12: Brian, M'
speaker_infos[420]='Caller 10: Brett, M'

speaker[1478]='Caller 11: Robyn, F'
speaker[2699]='Caller 23: Maureen, F'
speaker[3172]='Caller 9: Maureen, F'
speaker[4483]='Presenter 1b: Paul Murray, M' # maybe delete this file
speaker[8600]='Caller 12: Brian, M'
speaker[8858]='Caller 10: Brett, M'

In [20]:
# # using word_tokenize to separate out speaker info
speaker_toks=[]
for x in speaker:
    if len(x) > 5:
        speaker_toks.append(nltk.word_tokenize(x))

In [21]:
# speaker_toks['Caller', '33','b',':', 'Chris',',','male']
speaker_toks[:100]

[['Presenter', '1', ':', 'Simon', 'Marnie', ',', 'M'], ['Expert', '1', ':', 'Angus', 'Stewart', ',', 'M'], ['Caller', '1', ':', 'Suzanne', ',', 'F'], ['Caller', '2', ':', 'Lisa', ',', 'F'], ['Caller', '3', ':', 'Sally', ',', 'F'], ['Caller', '4', ':', 'Danny', ',', 'M'], ['Caller', '5', ':', 'Trevor', ',', 'M'], ['Caller', '6', ':', 'Gillian', ',', 'F'], ['Caller', '7', ':', 'Colleen', ',', 'F'], ['Caller', '8', ':', 'Bernie', ',', 'M'], ['Caller', '9', ':', 'William', ',', 'M'], ['Expert', '2', ':', 'Jeanne', 'Villani', ',', 'F'], ['Caller', '10', ':', 'Beth', ',', 'F'], ['Caller', '11', ':', 'Lynne', ',', 'F'], ['Caller', '12', ':', 'Jack', ',', 'M'], ['Presenter', '1', ':', 'Simon', 'Marnie', ',', 'M'], ['Expert', '1', ':', 'Les', ',', 'M'], ['Expert', '2', ':', 'Pete', ',', 'M'], ['Caller', '1', ':', 'Julie', ',', 'F'], ['Caller', '2', ':', 'Janelle', ',', 'F'], ['Caller', '3', ':', 'Geoff', ',', 'M'], ['Caller', '4', ':', 'Terry', ',', 'M'], ['Expert', '3', ':', 'John', 'Hall', ',

In [22]:
# # NOT A GOOD METHOD! only a good method for type

# # getting gender, type, and number
len(speaker_toks)
speaker_toks[:5]
# gen=[x[-1] for x in speaker_toks]
speaker_type=[x[0][:1] for x in speaker_toks]

len(speaker_type)
# len(gen)

428

[['Presenter', '1', ':', 'Simon', 'Marnie', ',', 'M'], ['Expert', '1', ':', 'Angus', 'Stewart', ',', 'M'], ['Caller', '1', ':', 'Suzanne', ',', 'F'], ['Caller', '2', ':', 'Lisa', ',', 'F'], ['Caller', '3', ':', 'Sally', ',', 'F']]

428

In [23]:
for x in speaker_infos:
    if x[-1:] != "F" and x[-1:] != "M":
        print(x)
        print(speaker_infos.index(x))
        
# Graham's gender: "" -> "NaN"
# Chris's gender: "male" -> "M"

Expert 1: Doctor Graham
111
Caller 33b: Chris, male
359


In [24]:
speaker.index("Expert 1: Doctor Graham")
speaker.index("Caller 33b: Chris, male")

speaker[2967]
speaker[7659]

2967

7659

'Expert 1: Doctor Graham'

'Caller 33b: Chris, male'

In [25]:
speaker_infos[111] = "Expert 1: Doctor Graham, NAN"
speaker_infos[359] = "Caller 33b: Chris, M"

speaker[2967] = "Expert 1: Doctor Graham, NAN"
speaker[7659] = "Caller 33b: Chris, M"

In [26]:
# Correctly adjusted the incorrect gender formats for Graham and Chris!!

speaker_infos[111]
speaker_infos[359]

speaker[2967]
speaker[7659]

'Expert 1: Doctor Graham, NAN'

'Caller 33b: Chris, M'

'Expert 1: Doctor Graham, NAN'

'Caller 33b: Chris, M'

In [27]:
# gen

# # fixing Graham
# speaker_infos[gen.index("Graham")] # does not list a gender
# gen[gen.index("Graham")]="NAN"

# # fixing "male" for gender IN speaker_infos AND speaker
# speaker_infos[gen.index("male")]
# speaker_infos[gen.index("male")]='Caller 33b: Chris, M'
# print("index of 'male':",str(gen.index("male")))

# # this claims chris is an Expert - WRONG 
# speaker[gen.index("male")]

# len(gen)
# len(speaker)
# len(speaker_infos)

# # CANNOT DO THIS BECAUSE SPEAKER'S INDEX IS NOT THE SAME AS GEN'S INDEX
# # speaker[gen.index("male")]='Caller 33b: Chris, M'       
#     # THIS IS WHY 33B WAS IN ABCE1!!!! 

# speaker[speaker.index('Caller 33b: Chris, male')]='Caller 33b: Chris, M'
# speaker_infos[gen.index("male")]

# gen[gen.index("male")]='M'
# gen

In [28]:
# speaker_infos.index('Caller 33b: Chris, M') # 359
# speaker.index('Caller 33b: Chris, M') # 7659

# speaker_num

# line_fid[359]
# line_fid[7659]


# speaker_infos[:20]
# speaker_num[:20]

# # used to see a ":" as the number for Jenny, Caller 5
#     # ex: Jenny's first line is called "C5" instead of Caller 5
#     # see below for fixes

In [29]:
# # which speaker_num items are wrong?
# count=0
# for x in speaker_num:
#     if x == ":":
#         print(x)
#         speaker_infos[count]
#         print(count)
#     count+=1
    
# # later rebuild speaker_num using regex
# # to find these where the 1st instance uses C,E,or P instead of Caller, Expert etc, can use .startswith

In [30]:
# speakers whose words are not written out!

for x in speaker_infos:
    if x[2].isalpha()== False:
        print(x)
        print(speaker_infos.index(x))

C5: Jenny, F
105
C14: Noelene, F
244


In [31]:
# fixing the above lines that labeled the number as a colon

# speaker_num[105] = '5'
# speaker_num[244] = '14'

# Fixing Jenny and Noelene!!
speaker_infos[105]='Caller 5: Jenny, F'
speaker_infos[244]='Caller 14: Noelene, F'

speaker.index('C5: Jenny, F') # 2834
speaker.index('C14: Noelene, F') # 6063

speaker[2834] = 'Caller 5: Jenny, F'
speaker[6063] = 'Caller 14: Noelene, F'

2834

6063

In [32]:
# Jenny and Noelene are fixed!

speaker_infos[105]
speaker_infos[244]

speaker[2834]
speaker[6063]

'Caller 5: Jenny, F'

'Caller 14: Noelene, F'

'Caller 5: Jenny, F'

'Caller 14: Noelene, F'

In [33]:
# # dictionary for gender of the speaker from speaker info
# gen_dict={x:x[-1] for x in speaker_infos}

In [34]:
# gen_dict['Presenter 1: Simon Marnie, M']

In [35]:
# # dictionary for filename from speaker info
# file_speaker_dict={x:line_fid[speaker.index(x)] for x in speaker_infos}

In [36]:
# file_speaker_dict['Presenter 1: Simon Marnie, M']
# file_speaker_dict['Caller 12: Brian, M']

In [37]:
# # dictionary for speaker's filename, gender, and shortcut label (ex: P1, C12, etc..)
# gen_file_speaker_dict={x:[line_fid[speaker.index(x)],x[-1],speaker_type[speaker_infos.index(x)]+speaker_num[speaker_infos.index(x)]] for x in speaker_infos}

In [38]:
# gen_file_speaker_dict['Presenter 1: Simon Marnie, M']
# gen_file_speaker_dict['Caller 12: Brian, M']

In [39]:
# Figuring out how to do the regular expression to find the speaker_number
# USE x IN THE BIG LOOP
import re

s="Presenter 1: Simon Marnie, M"
s="Caller 33b: Chris, M"

re.findall('\d+[a-z]*', s)[0]


# I have now commented out the speaker number list -- working on removing it now!!
# I can fix the speaker number list later by doing the above method! I can also potentially delete that list? superfluous?

'33b'

In [40]:
# using regular expressions to find the speaker num!
for x in speaker_infos:
    speaker_num=[re.findall('\d+[a-z]*', x)[0] for x in speaker_infos]
    
len(speaker_num)

428

In [41]:
# creating a lists for all speakers and unique speakers

new_speaker_list=[]    # a list of each line's speaker with the filename
all_speaker_type=[]    # a list of each line's speaker's P/C/E designation
unique_speaker=[]      # a list of each speaker's unique id (ex: ABCE1-P1)
unique_speaker_type=[] # a list of each unique speaker's type (P/C/E)
unique_speaker_fid=[]  # a list of which file contains each unique speaker

count = 0 
all_count=0

used_speaker=[]

# for x in speaker[:500]:
for x in speaker:
    
    x_loc=speaker.index(x)
#     print(line_fid[x_loc])
#     print(line_fid[x_loc].index("-"))
#     print(line_fid[x_loc][:dash+1])
    dash=line_fid[x_loc].index("-")
    
    
    # adjusts for people like Simon Marnie who show up in multiple files (ABCE1 and ABCE2)
    if x in used_speaker:
        x_loc=all_count
        dash=line_fid[all_count].index("-")
    
    if x in speaker_infos:
        
#         print(speaker_type[speaker_infos.index(x)])
#         print(speaker_num[speaker_infos.index(x)])
#         print(line_fid[x_loc][:dash+1])

        current_speaker=speaker_infos[count]
        new_speaker_list.append(line_fid[x_loc][:dash+1]+speaker_type[speaker_infos.index(x)]+speaker_num[speaker_infos.index(x)])
        all_speaker_type.append(speaker_type[speaker_infos.index(x)])
        
        # ONLY CHECK THESE IF YOU LIMIT THE RANGE OF speaker!
#         print(line_fid[x_loc][:dash+1]) # CORRECT
#         print(speaker_type[count]) # GIVES WRONG LETTER -- CANNOT USE COUNT FEATURE
#         print(x[0])
#         print(x+"\n") # correctly identifies Marnie's second occurance as being in ABCE2

        NUM=re.findall('\d+[a-z]*', x) # need regex to fix the number - get type from first letter of x instead of using speaker_type list
        
        
        # NAT4 Duplicate Loop:
            # DUPLICATES: File NAT4 begins with a new Presenter 1, Caller1-6 at the end of the file!!
        if line_fid[x_loc][:dash+1]+x[0]+NUM[0] in unique_speaker:
            
            print(line_fid[x_loc][:dash+1]+x[0]+NUM[0])
                # The second half of NAT4 will now be NAT4.5
                
            unique_speaker.append(line_fid[x_loc][:dash]+".5"+line_fid[x_loc][dash-1:dash+1]+x[0]+NUM[0])
            unique_speaker_type.append(x[0])
            unique_speaker_fid.append(line_fid[x_loc][:dash]+".5"+line_fid[x_loc][dash-1:dash+1])
            
        else:
            unique_speaker.append(line_fid[x_loc][:dash+1]+x[0]+NUM[0])
            unique_speaker_type.append(x[0])
            unique_speaker_fid.append(line_fid[x_loc][:dash+1]) # refers to their first line to find the file id
        
#         unique_speaker.append(line_fid[x_loc][:dash+1]+x[0]+NUM[0])
#         unique_speaker_type.append(x[0])
#         unique_speaker_fid.append(line_fid[x_loc][:dash+1]) # refers to their first line to find the file id
        
        count+=1
        used_speaker.append(x)
    else:
        new_speaker_list.append(line_fid[x_loc][:dash+1]+x)
        all_speaker_type.append(x[0]) # only want the letter
    
    all_count+=1

NAT4-P1
NAT4-C1
NAT4-C2
NAT4-C3
NAT4-C4
NAT4-C5
NAT4-C6


In [42]:
len(used_speaker)
len(unique_speaker) # 428

# After adding the NAT4 Duplicate Loop, every speaker has a unique ID:
len(set(unique_speaker)) # 421 -> 428 

428

428

428

In [43]:
print()

In [44]:
# how many times is Simon Marnie the main presenter? Answer: twice!
count=0
for x in speaker:
    if x == "Presenter 1: Simon Marnie, M":
        count
    count+=1
print("First instance of Simon Marnie:",str(speaker[0]), ",", str(line_fid[0]))
print("Second instance of Simon Marnie:",str(speaker[471]), ",", str(line_fid[471]))

# line_fid WAS correctly built (at least where Simon Marnie is concnered)
    # Simon is in ABCE1 and ABCE2

0

471

First instance of Simon Marnie: Presenter 1: Simon Marnie, M , ABCE1-raw.txt
Second instance of Simon Marnie: Presenter 1: Simon Marnie, M , ABCE2-raw.txt


In [45]:
# comparing new_speaker_list, speaker, and all_speaker_type to make sure they are the same

# 9026 lines
len(new_speaker_list)
len(speaker)
len(all_speaker_type)

print("Original list of speakers for each line:")
speaker[:100]
speaker[-100:]

print("New list of speakers for each line:")
new_speaker_list[:100]
new_speaker_list[-100:]

print("List of the type of speaker for each line:")
all_speaker_type[:100]
all_speaker_type[-100:]

print("List of fileid for each line:")
line_fid[:100]
line_fid[-100:]

9026

9026

9026

Original list of speakers for each line:


['Presenter 1: Simon Marnie, M', 'Expert 1: Angus Stewart, M', 'P1', 'E1', 'P1', 'P1', 'Caller 1: Suzanne, F', 'P1', 'C1', 'P1', 'C1', 'P1', 'C1', 'P1', 'C1', 'P1', 'E1', 'C1', 'E1', 'C1', 'E1', 'C1', 'E1', 'C1', 'E1', 'C1', 'E1', 'C1', 'E1', 'C1', 'E1', 'C1', 'E1', 'C1', 'E1', 'C1', 'P1', 'C1', 'P1', 'C1', 'P1', 'E1', 'P1', 'E1', 'P1', 'E1', 'P1', 'E1', 'P1', 'E1', 'P1', 'E1', 'P1', 'P1', 'Caller 2: Lisa, F', 'P1', 'C2', 'P1', 'C2', 'E1', 'C2', 'E1', 'P1', 'C2', 'P1', 'C2', 'P1', 'E1', 'C2', 'E1', 'C2', 'E1', 'C2', 'E1', 'P1', 'E1', 'C2', 'P1', 'C2', 'E1', 'P1', 'C2', 'P1', 'C2', 'P1', 'P1', 'Caller 3: Sally, F', 'P1', 'C3', 'P1', 'C3', 'P1', 'C3', 'E1', 'C3', 'P1', 'C3', 'P1', 'E1', 'P1']

['P1', 'C13', 'P1', 'C13', 'P1', 'C13', 'P1', 'C13', 'P1', 'C13', 'P1', 'C13', 'P1', 'C13', 'P1', 'C13', 'P1', 'Caller 14: Ted, M', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'C14', 'P1', 'P1', 'P1', 'Caller 15: Britney, F', 'P1', 'C15', 'P1', 'C15', 'P1', 'C15', 'P1', 'C15', 'P1', 'C15', 'P1', 'C15', 'P1', 'C15', 'P1', 'Caller 16: Jesse, M', 'P1', 'C16', 'P1', 'C16', 'P1', 'C16', 'P1', 'C16', 'P1', 'C16', 'P1', 'C16', 'P1', 'Caller 17: Kieran, M', 'P1', 'C17']

New list of speakers for each line:


['ABCE1-P1', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-P1', 'ABCE1-C1', 'ABCE1-P1', 'ABCE1-C1', 'ABCE1-P1', 'ABCE1-C1', 'ABCE1-P1', 'ABCE1-C1', 'ABCE1-P1', 'ABCE1-C1', 'ABCE1-P1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-P1', 'ABCE1-C1', 'ABCE1-P1', 'ABCE1-C1', 'ABCE1-P1', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-P1', 'ABCE1-C2', 'ABCE1-P1', 'ABCE1-C2', 'ABCE1-P1', 'ABCE1-C2', 'ABCE1-E1', 'ABCE1-C2', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-C2', 'ABCE1-P1', 'ABCE1-C2', 'ABCE1-P1', 'ABCE1-E1', 'ABCE1-C2', 'ABCE1-E1', 'ABCE1-C2', 'ABCE1-E1', 'ABCE1-C2', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-E1', 'ABCE1-C2', 'ABCE1-P1', 'ABCE1-C2', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-C2', 'ABCE1-P1', 'AB

['ABCE1-P1', 'COME1-C13', 'ABCE1-P1', 'COME1-C13', 'ABCE1-P1', 'COME1-C13', 'ABCE1-P1', 'COME1-C13', 'ABCE1-P1', 'COME1-C13', 'ABCE1-P1', 'COME1-C13', 'ABCE1-P1', 'COME1-C13', 'ABCE1-P1', 'COME1-C13', 'ABCE1-P1', 'NAT8-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'COME1-C14', 'ABCE1-P1', 'ABCE1-P1', 'ABCE1-P1', 'NAT8-C15', 'ABCE1-P1', 'COME1-C15', 'ABCE1-P1', 'COME1-C15', 'ABCE1-P1', 'COME1-C15', 'ABCE1-P1', 'COME1-C15', 'ABCE1-P1', 'COME1-C15', 'ABCE1-P1', 'COME1-C15', 'A

List of the type of speaker for each line:


['P', 'E', 'P', 'E', 'P', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'E', 'C', 'E', 'C', 'E', 'C', 'E', 'C', 'E', 'C', 'E', 'C', 'E', 'C', 'E', 'C', 'E', 'C', 'E', 'C', 'P', 'C', 'P', 'C', 'P', 'E', 'P', 'E', 'P', 'E', 'P', 'E', 'P', 'E', 'P', 'E', 'P', 'P', 'C', 'P', 'C', 'P', 'C', 'E', 'C', 'E', 'P', 'C', 'P', 'C', 'P', 'E', 'C', 'E', 'C', 'E', 'C', 'E', 'P', 'E', 'C', 'P', 'C', 'E', 'P', 'C', 'P', 'C', 'P', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'E', 'C', 'P', 'C', 'P', 'E', 'P']

['P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'P', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C', 'P', 'C']

List of fileid for each line:


['ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.txt', 'ABCE1-raw.tx

['NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-raw.txt', 'NAT8-r

In [46]:
# checking unique_speaker, unique_speaker_type, and unique_speaker_fid

# 428 for each
len(speaker_infos)
len(unique_speaker)
len(unique_speaker_type)
len(unique_speaker_fid)

print("Speaker Infos:")
speaker_infos[:100]
speaker_infos[-100:]

print("Unique Speaker ID:")
unique_speaker[:100]
unique_speaker[-100:]

print("Unique Speaker Type:")
unique_speaker_type[:100] # Note: NOT made from speaker_type
unique_speaker_type[-100:]

print("Unique Speaker Fileid:")
unique_speaker_fid[:100]
unique_speaker_fid[-100:]

428

428

428

428

Speaker Infos:


['Presenter 1: Simon Marnie, M', 'Expert 1: Angus Stewart, M', 'Caller 1: Suzanne, F', 'Caller 2: Lisa, F', 'Caller 3: Sally, F', 'Caller 4: Danny, M', 'Caller 5: Trevor, M', 'Caller 6: Gillian, F', 'Caller 7: Colleen, F', 'Caller 8: Bernie, M', 'Caller 9: William, M', 'Expert 2: Jeanne Villani, F', 'Caller 10: Beth, F', 'Caller 11: Lynne, F', 'Caller 12: Jack, M', 'Presenter 1: Simon Marnie, M', 'Expert 1: Les, M', 'Expert 2: Pete, M', 'Caller 1: Julie, F', 'Caller 2: Janelle, F', 'Caller 3: Geoff, M', 'Caller 4: Terry, M', 'Expert 3: John Hall, M', 'Caller 5: Judy, F', 'Caller 6: Dawn, F', 'Caller 7: Paul, M', 'Caller 8: Linda, F', 'Caller 9: Croydon, M', 'Caller 10: Joan, F', 'Presenter 1: Lynne Haultain, F', 'Presenter 2: Jurate Sasnaitis, F', 'Caller 1: Rick, M', 'Caller 2: Sarah, F', 'Caller 3: Liz, F', 'Caller 4: Cathy, F', 'Caller 6: Bullia, F', 'Caller 7: Juliet, F', 'Caller 8: Melanie, F', 'Caller 9: Margaret, F', 'Caller 10: Lisa, F', 'Caller 11: Robyn, F', 'Caller 12: Mary,

['Caller 4: Joan, F', 'Caller 5: Judy, F', 'Caller 6: Lynne, F', 'Caller 7: Edith, F', 'Caller 8: Karen, F', 'Caller 9: Ruth, F', 'Caller 10: Wayne, M', 'Caller 11: David, M', 'Caller 12: Helen, F', 'Caller 13: Graham, M', 'Caller 14: Sue, F', 'Caller 15: Jennifer, F', 'Caller 16: Greg, M', 'Caller 17: Kevin, M', 'Caller 18: Glenys, F', 'Caller 19: June, F', 'Caller 20: Betty, F', 'Caller 21: Jeremiah, M', 'Caller 22: Michael, M', 'Caller 23: Ken, M', 'Caller 24: Jim, M', 'Caller 25: Debbie, F', 'Caller 26: Frank, M', 'Caller 27: Sonia, F', 'Caller 28: Sean, M', 'Caller 29: Anne, F', 'Caller 30: Lesley, F', 'Caller 31: Ron, M', 'Caller 32a: Sue, F', 'Caller 32b: Mary, F', 'Caller 33a: Gerrard, M', 'Caller 33b: Chris, M', 'Caller 34: Keira, F', 'Caller 35: Robin, M', 'Caller 36: Wayne, M', 'Caller 37: Sue, F', 'Caller 38: Jill, F', 'Caller 39: Peter, M', 'Caller 40: Rick, M', 'Caller 41: Jean Pierre, M', 'Caller 42: Magpie, F', 'Caller 43: Warren, M', 'Caller 44: Colin, M', 'Caller 45: 

Unique Speaker ID:


['ABCE1-P1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-C2', 'ABCE1-C3', 'ABCE1-C4', 'ABCE1-C5', 'ABCE1-C6', 'ABCE1-C7', 'ABCE1-C8', 'ABCE1-C9', 'ABCE1-E2', 'ABCE1-C10', 'ABCE1-C11', 'ABCE1-C12', 'ABCE2-P1', 'ABCE2-E1', 'ABCE2-E2', 'ABCE2-C1', 'ABCE2-C2', 'ABCE2-C3', 'ABCE2-C4', 'ABCE2-E3', 'ABCE2-C5', 'ABCE2-C6', 'ABCE2-C7', 'ABCE2-C8', 'ABCE2-C9', 'ABCE2-C10', 'ABCE3-P1', 'ABCE3-P2', 'ABCE3-C1', 'ABCE3-C2', 'ABCE3-C3', 'ABCE3-C4', 'ABCE3-C6', 'ABCE3-C7', 'ABCE3-C8', 'ABCE3-C9', 'ABCE3-C10', 'ABCE3-C11', 'ABCE3-C12', 'ABCE4-E1', 'ABCE4-P1', 'ABCE4-C1', 'ABCE4-C2', 'ABCE4-C3', 'ABCE4-C4', 'ABCE4-C5', 'ABCE4-C6', 'ABCE4-C7', 'ABCNE1-P1', 'ABCNE1-E1', 'ABCNE1-C1', 'ABCNE1-C2', 'ABCNE1-C3', 'ABCNE1-C4', 'ABCNE1-C5', 'ABCNE1-C6', 'ABCNE1-C7', 'ABCNE1-C8', 'ABCNE1-C9', 'ABCNE2-P1', 'ABCNE2-E1', 'ABCNE2-C1', 'ABCNE2-C2', 'ABCNE2-C3', 'ABCNE2-C4', 'ABCNE2-C5', 'ABCNE2-C6', 'ABCNE2-C7', 'ABCNE2-C8', 'ABCNE2-C9', 'COME1-P1', 'COME1-E1', 'COME1-C1', 'COME1-C2', 'COME1-C3', 'COME1-C4', 'COME1-C5', 'COME1-C6'

['NAT4-C4', 'NAT4-C5', 'NAT4-C6', 'NAT4-C7', 'NAT4-C8', 'NAT4-C9', 'NAT4-C10', 'NAT4-C11', 'NAT4-C12', 'NAT4-C13', 'NAT4-C14', 'NAT4-C15', 'NAT4-C16', 'NAT4-C17', 'NAT4-C18', 'NAT4-C19', 'NAT4-C20', 'NAT4-C21', 'NAT4-C22', 'NAT4-C23', 'NAT4-C24', 'NAT4-C25', 'NAT4-C26', 'NAT4-C27', 'NAT4-C28', 'NAT4-C29', 'NAT4-C30', 'NAT4-C31', 'NAT4-C32a', 'NAT4-C32b', 'NAT4-C33a', 'NAT4-C33b', 'NAT4-C34', 'NAT4-C35', 'NAT4-C36', 'NAT4-C37', 'NAT4-C38', 'NAT4-C39', 'NAT4-C40', 'NAT4-C41', 'NAT4-C42', 'NAT4-C43', 'NAT4-C44', 'NAT4-C45', 'NAT4-C46', 'NAT4-C47', 'NAT4-C48', 'NAT4-C49', 'NAT4-C50', 'NAT4-C51', 'NAT4.54-P1', 'NAT4-E2', 'NAT4-E3', 'NAT4.54-C1', 'NAT4.54-C2', 'NAT4.54-C3', 'NAT4.54-C4', 'NAT4.54-C5', 'NAT4.54-C6', 'NAT6-P1', 'NAT6-E1', 'NAT6-C1', 'NAT6-C2', 'NAT6-C3', 'NAT6-C4', 'NAT6-C5', 'NAT6-C6', 'NAT6-C7', 'NAT7-P1', 'NAT7-C1', 'NAT7-C3', 'NAT7-C4', 'NAT7-C5', 'NAT7-C6', 'NAT7-C7', 'NAT7-C8', 'NAT7-C9', 'NAT7-E1', 'NAT7-C10', 'NAT7-C11', 'NAT7-C12', 'NAT7-C13', 'NAT7-C14', 'NAT8-P1', '

Unique Speaker Type:


['P', 'E', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'E', 'C', 'C', 'C', 'P', 'E', 'E', 'C', 'C', 'C', 'C', 'E', 'C', 'C', 'C', 'C', 'C', 'C', 'P', 'P', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'E', 'P', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'P', 'E', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'P', 'E', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'P', 'E', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'E', 'C', 'C', 'C', 'C', 'C', 'C', 'P']

['C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'P', 'E', 'E', 'C', 'C', 'C', 'C', 'C', 'C', 'P', 'E', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'P', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'E', 'C', 'C', 'C', 'C', 'C', 'P', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C', 'C']

Unique Speaker Fileid:


['ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE1-', 'ABCE2-', 'ABCE2-', 'ABCE2-', 'ABCE2-', 'ABCE2-', 'ABCE2-', 'ABCE2-', 'ABCE2-', 'ABCE2-', 'ABCE2-', 'ABCE2-', 'ABCE2-', 'ABCE2-', 'ABCE2-', 'ABCE3-', 'ABCE3-', 'ABCE3-', 'ABCE3-', 'ABCE3-', 'ABCE3-', 'ABCE3-', 'ABCE3-', 'ABCE3-', 'ABCE3-', 'ABCE3-', 'ABCE3-', 'ABCE3-', 'ABCE4-', 'ABCE4-', 'ABCE4-', 'ABCE4-', 'ABCE4-', 'ABCE4-', 'ABCE4-', 'ABCE4-', 'ABCE4-', 'ABCNE1-', 'ABCNE1-', 'ABCNE1-', 'ABCNE1-', 'ABCNE1-', 'ABCNE1-', 'ABCNE1-', 'ABCNE1-', 'ABCNE1-', 'ABCNE1-', 'ABCNE1-', 'ABCNE2-', 'ABCNE2-', 'ABCNE2-', 'ABCNE2-', 'ABCNE2-', 'ABCNE2-', 'ABCNE2-', 'ABCNE2-', 'ABCNE2-', 'ABCNE2-', 'ABCNE2-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-', 'COME1-

['NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4-', 'NAT4.54-', 'NAT4-', 'NAT4-', 'NAT4.54-', 'NAT4.54-', 'NAT4.54-', 'NAT4.54-', 'NAT4.54-', 'NAT4.54-', 'NAT6-', 'NAT6-', 'NAT6-', 'NAT6-', 'NAT6-', 'NAT6-', 'NAT6-', 'NAT6-', 'NAT6-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT7-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-', 'NAT8-']

In [47]:
# Check Marnie in the new lists:

# marnie should be ABCE2
speaker_infos[15]
print("Marnie's 2ND file Unique ID:", str(unique_speaker[15]))
print("Marnie's speaker type:",str(unique_speaker_type[15]))
print("Marnie's fileid:",str(unique_speaker_fid[15]))

'Presenter 1: Simon Marnie, M'

Marnie's 2ND file Unique ID: ABCE2-P1
Marnie's speaker type: P
Marnie's fileid: ABCE2-


In [48]:
# finding extra items in the text like laughter and interruptions

# THIS IS A WORK IN PROGRESS
# I will change my methods to use regular expressions in order to find backchannels instead of using all of these loops

# extras=[]
# all_sq_braqs=[]
# all_carrots=[]

# for line in text:
    
#     line_extras=[]
#     sq_bra_each_line=[]
#     car_each_line=[]
    
#     # skipping lines that are not speech 
#     if line.startswith("\r\n{")==False:
        
#         if "<" in line: # others' speech, laughter, "inaudible"
#             CARROTS_INDEX=[]
#             END_CARROTS_INDEX=[]
#             for y in line:
#                 if y == "<":
#                     CARROTS_INDEX.append(line.index(y))
#                 elif y == ">":
#                     END_CARROTS_INDEX.append(line.index(y))
#             for z in CARROTS_INDEX:
#                 car=line[z+1:END_CARROTS_INDEX[CARROTS_INDEX.index(z)]]
#                 car_each_line.append(car)

#         else:
#             car_each_line.append("Nan")
            
#         all_carrots.append(car_each_line)        
     
#         if "{" in line: # spelling corrections, interruptions to the program (I tried to remove previously)
#             BRAQ_INDEX=[]
#             END_SQ_BRAQ_INDEX=[]
#             for y in line:
#                 if y == "{":
#                     BRAQ_INDEX.append(line.index(y))
#                 elif y == "}":
#                     END_SQ_BRAQ_INDEX.append(line.index(y))
#             for z in BRAQ_INDEX:
#                 sq_bra=line[z+1:END_SQ_BRAQ_INDEX[BRAQ_INDEX.index(z)]]
#                 sq_bra_each_line.append(sq_bra)
                
#         else: 
#             sq_bra_each_line.append("Nan")
            
# #         print(sq_bra_each_line)
#         all_sq_braqs.append(sq_bra_each_line)
# #         print(all_sq_braqs)
        
# #         if "[" in line:
# #             bracket=line.index("[")
# #             if "]" not in line:         
# #                 print(line)
# #                 print(text.index(line))    # 3093
# #             end_bracket=line.index("]")
# #             bra=line[bracket:end_bracket]
# #             line_extras.append(bra)

# #         if "[" not in line and "{" not in line and "<" not in line:


In [49]:
# all_sq_braqs[:10]
# text[:10]

In [50]:
# double checking which lists are which for my next loop: finding the number of turns per speaker
len(line_fid)
len(utterance_num)
len(new_speaker_list)
len(unique_speaker)

# these 2 should not line up, just taking a peak at both
print("Unique Speaker List: length",str(len(unique_speaker)))
unique_speaker[:10]
print("New Speaker List for each line: length",str(len(new_speaker_list)))
new_speaker_list[:10]

9026

9026

9026

428

Unique Speaker List: length 428


['ABCE1-P1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-C2', 'ABCE1-C3', 'ABCE1-C4', 'ABCE1-C5', 'ABCE1-C6', 'ABCE1-C7', 'ABCE1-C8']

New Speaker List for each line: length 9026


['ABCE1-P1', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-E1', 'ABCE1-P1', 'ABCE1-P1', 'ABCE1-C1', 'ABCE1-P1', 'ABCE1-C1', 'ABCE1-P1']

In [51]:
'NAT4-C33b' in unique_speaker
# successfully got the letters to work!

True

In [52]:
# Finding number of turns per speaker
speaker_counts={}
speaker_counts_L=[]
for x in unique_speaker:
    count=0

    for y in new_speaker_list: # remember: new speaker list is the new speaker list for each line of the data
        if x==y:
            count+=1
    speaker_counts[x]=count
    speaker_counts_L.append(int(count))

In [53]:
len(unique_speaker)

len(speaker_counts)  # why 6 short?
len(speaker_counts_L)

# because there are duplicates in unique_speaker!!!
len(set(unique_speaker)) #length is 421 instead of 428
# This means that Simon Marnie is not the only duplicate speaker. I need to 
# analyze unique_speaker to find duplicates and adjust my loops from above. 

speaker_counts_L[:100]
# Originally two people talked 2678 times, I fixed that issue

len(speaker_infos)

unique_speaker[:100]

unique_speaker[0]
unique_speaker[15] # Simon Marnie is P1 for ABCE1 and ABCE2
speaker_infos[0]
speaker_infos[15] 

speaker_infos[:17]
unique_speaker[:17]


428

428

428

428

[2677, 1955, 203, 212, 202, 184, 166, 165, 189, 149, 230, 342, 166, 113, 85, 1, 1, 1, 1, 1, 1, 1, 45, 1, 1, 1, 1, 1, 1, 1, 708, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 77, 85, 69, 79, 57, 1, 54, 34, 10, 13, 24, 45, 1]

428

['ABCE1-P1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-C2', 'ABCE1-C3', 'ABCE1-C4', 'ABCE1-C5', 'ABCE1-C6', 'ABCE1-C7', 'ABCE1-C8', 'ABCE1-C9', 'ABCE1-E2', 'ABCE1-C10', 'ABCE1-C11', 'ABCE1-C12', 'ABCE2-P1', 'ABCE2-E1', 'ABCE2-E2', 'ABCE2-C1', 'ABCE2-C2', 'ABCE2-C3', 'ABCE2-C4', 'ABCE2-E3', 'ABCE2-C5', 'ABCE2-C6', 'ABCE2-C7', 'ABCE2-C8', 'ABCE2-C9', 'ABCE2-C10', 'ABCE3-P1', 'ABCE3-P2', 'ABCE3-C1', 'ABCE3-C2', 'ABCE3-C3', 'ABCE3-C4', 'ABCE3-C6', 'ABCE3-C7', 'ABCE3-C8', 'ABCE3-C9', 'ABCE3-C10', 'ABCE3-C11', 'ABCE3-C12', 'ABCE4-E1', 'ABCE4-P1', 'ABCE4-C1', 'ABCE4-C2', 'ABCE4-C3', 'ABCE4-C4', 'ABCE4-C5', 'ABCE4-C6', 'ABCE4-C7', 'ABCNE1-P1', 'ABCNE1-E1', 'ABCNE1-C1', 'ABCNE1-C2', 'ABCNE1-C3', 'ABCNE1-C4', 'ABCNE1-C5', 'ABCNE1-C6', 'ABCNE1-C7', 'ABCNE1-C8', 'ABCNE1-C9', 'ABCNE2-P1', 'ABCNE2-E1', 'ABCNE2-C1', 'ABCNE2-C2', 'ABCNE2-C3', 'ABCNE2-C4', 'ABCNE2-C5', 'ABCNE2-C6', 'ABCNE2-C7', 'ABCNE2-C8', 'ABCNE2-C9', 'COME1-P1', 'COME1-E1', 'COME1-C1', 'COME1-C2', 'COME1-C3', 'COME1-C4', 'COME1-C5', 'COME1-C6'

'ABCE1-P1'

'ABCE2-P1'

'Presenter 1: Simon Marnie, M'

'Presenter 1: Simon Marnie, M'

['Presenter 1: Simon Marnie, M', 'Expert 1: Angus Stewart, M', 'Caller 1: Suzanne, F', 'Caller 2: Lisa, F', 'Caller 3: Sally, F', 'Caller 4: Danny, M', 'Caller 5: Trevor, M', 'Caller 6: Gillian, F', 'Caller 7: Colleen, F', 'Caller 8: Bernie, M', 'Caller 9: William, M', 'Expert 2: Jeanne Villani, F', 'Caller 10: Beth, F', 'Caller 11: Lynne, F', 'Caller 12: Jack, M', 'Presenter 1: Simon Marnie, M', 'Expert 1: Les, M']

['ABCE1-P1', 'ABCE1-E1', 'ABCE1-C1', 'ABCE1-C2', 'ABCE1-C3', 'ABCE1-C4', 'ABCE1-C5', 'ABCE1-C6', 'ABCE1-C7', 'ABCE1-C8', 'ABCE1-C9', 'ABCE1-E2', 'ABCE1-C10', 'ABCE1-C11', 'ABCE1-C12', 'ABCE2-P1', 'ABCE2-E1']

In [54]:
speaker_counts_L[-100:]

[2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 4, 1, 2, 4, 5, 4, 7, 4, 19, 2, 6, 6, 7, 1, 6, 4, 1, 3, 5, 8, 5, 3, 3, 4, 4, 5, 4, 4, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]

## Creating DataFrames
### Text DataFrame

In [55]:
# Text DataFrame
art_df = pd.DataFrame(new_speaker_list)
art_df.head()

art_df.columns = ["Speaker"]
art_df.head()

0
0  ABCE1-P1
1  ABCE1-E1
2  ABCE1-P1
3  ABCE1-E1
4  ABCE1-P1

Speaker
0  ABCE1-P1
1  ABCE1-E1
2  ABCE1-P1
3  ABCE1-E1
4  ABCE1-P1

In [56]:
art_df["Utterance_Number"]=pd.DataFrame(utterance_num)
art_df["Text"]=pd.DataFrame(text)
art_df.head()
art_df.tail()

Speaker  Utterance_Number  \
0  ABCE1-P1                 1   
1  ABCE1-E1                 2   
2  ABCE1-P1                 3   
3  ABCE1-E1                 4   
4  ABCE1-P1                 5   

                                                Text  
0  Thanks for that John Hall now John Hall will b...  
1                        I guess yeah yeah <laughs>.  
2  He's also known <E1 sounds reasonable> for his...  
3                                              Okay.  
4  Jeanne Villani does and we'll find out the sec...

Speaker  Utterance_Number  \
9021  COME1-C16               359   
9022   ABCE1-P1               360   
9023   NAT8-C17               361   
9024   ABCE1-P1               362   
9025  COME1-C17               363   

                                                   Text  
9021                                   Thanks for that.  
9022  Uh let's see if we can just squeeze in one mor...  
9023                 Um okay yeah thanks for that Gaby.  
9024                                    That's alright.  
9025  Um and now that last caller I was just gunna s...

In [57]:
# how to do multi-level indexing
help(art_df.set_index)

Help on method set_index in module pandas.core.frame:

set_index(keys, drop=True, append=False, inplace=False, verify_integrity=False) method of pandas.core.frame.DataFrame instance
    Set the DataFrame index (row labels) using one or more existing
    columns. By default yields a new object.
    
    Parameters
    ----------
    keys : column label or list of column labels / arrays
    drop : boolean, default True
        Delete columns to be used as the new index
    append : boolean, default False
        Whether to append columns to existing index
    inplace : boolean, default False
        Modify the DataFrame in place (do not create a new object)
    verify_integrity : boolean, default False
        Check the new index for duplicates. Otherwise defer the check until
        necessary. Setting to False will improve the performance of this
        method
    
    Examples
    --------
    >>> df = pd.DataFrame({'month': [1, 4, 7, 10],
    ...                    'year': [2012, 20

In [58]:
# Sets the index to the speaker and utterance number
art_df = art_df.set_index(keys=["Speaker","Utterance_Number"])
art_df.head()
art_df.tail()

Text
Speaker  Utterance_Number                                                   
ABCE1-P1 1                 Thanks for that John Hall now John Hall will b...
ABCE1-E1 2                                       I guess yeah yeah <laughs>.
ABCE1-P1 3                 He's also known <E1 sounds reasonable> for his...
ABCE1-E1 4                                                             Okay.
ABCE1-P1 5                 Jeanne Villani does and we'll find out the sec...

Text
Speaker   Utterance_Number                                                   
COME1-C16 359                                                Thanks for that.
ABCE1-P1  360               Uh let's see if we can just squeeze in one mor...
NAT8-C17  361                              Um okay yeah thanks for that Gaby.
ABCE1-P1  362                                                 That's alright.
COME1-C17 363               Um and now that last caller I was just gunna s...

### Speaker DataFrame

In [59]:
# Speaker DataFrame
speaker_df = pd.DataFrame(unique_speaker)
speaker_df.head()

speaker_df.columns = ["Filename_Speaker"]
speaker_df.head()

0
0  ABCE1-P1
1  ABCE1-E1
2  ABCE1-C1
3  ABCE1-C2
4  ABCE1-C3

Filename_Speaker
0         ABCE1-P1
1         ABCE1-E1
2         ABCE1-C1
3         ABCE1-C2
4         ABCE1-C3

In [60]:
speaker_df["Speaker_Type"]=pd.DataFrame(unique_speaker_type) # is this right? investigate later
speaker_df["Gender"]=pd.DataFrame(gen)
speaker_df["Turns"] = pd.DataFrame(speaker_counts_L)
speaker_df.head()

NameError: name 'gen' is not defined

In [ ]:
speaker_df = speaker_df.set_index("Filename_Speaker")

In [ ]:
speaker_df.head()

# I have filename_speaker, speaker type, and gender
# I added number of lines or "turns" <-- this might need editing

In [ ]:
# help(speaker_df)
speaker_df.info()

In [ ]:
speaker_df.head(20)

In [ ]:
len(speaker_counts_L)

In [ ]:
# dataframe of presenters
P_df=speaker_df.loc[speaker_df["Speaker_Type"]=='P',:]
P_df.head()

In [ ]:
# dataframe of callers
C_df=speaker_df.loc[speaker_df["Speaker_Type"]=='C',:]
C_df.head()

In [ ]:
# dataframe of experts
E_df=speaker_df.loc[speaker_df["Speaker_Type"]=='E',:]
E_df.head()

In [ ]:
P_df.info()
C_df.info()
E_df.info()

# There are a lot more Callers than there are Presenters and Experts, but there are about equal numbers of Presenters and Experts

In [ ]:
# Looking at the Presenter DF
P_df.max() # 2677
P_df.min() # 1
P_df.sum() # 3490
# help(P_df["Turns"])

In [ ]:
P_df.describe()
C_df.describe()
E_df.describe()

There are a lot more callers than there are presenters and experts.
This means that the presenters and experts are probably more comparable. However, this will depend on the number of words that are actually uttered by the presenters vs. experts.
On average, presenters take about 109.06 turns. However, there is a very high standard deviation of 485. I believe that this is
because Simon Marnie talks a lot and is in 2 files. Experts had on average 65.97 turns, but also had a high standard deviation of about 329. 
The max for presenters (Simon Marnie) was 2677 and for experts it was 1955. These are outliers and are skewing the data.
In radio show talks, callers typically are only on the air for a minute or two. This explains why there are so many callers
with such low numbers of turns. On average, callers had almost 9 turns. The standard deviation is much lower, partially due to the large number of callers. The max, 230, still seems very high for callers and this may also be an outlier. 


Based on this corpus, the presenters (the show's hosts) take the most number of turns throughout the show.


As for who actually talks the most, I will be looking into that in my next analsis, which will involve word and sentence tokens.


Similarly, I will do an analysis on vocabulary size and sentence length to compare across types of speakers